archiver/helpers.py

```diff
@@ -317,6 +317,28 @@ def filename_without_extensions(path):
     return path.name[:-len(suffixes_string)]
 
 
+def filepath_without_extensions(path:Path) -> Path:
+    """Removes every suffix, including .partX"""
+    suffixes_string = "".join(path.suffixes)
+
+    return path.parent / path.name[:-len(suffixes_string)]
+
+def infer_source_name(source_path: Path) -> Path:
+
+    if not source_path.is_dir():
+        return filepath_without_extensions(source_path)
+    else:
+        all_files = [p for p in source_path.iterdir() if p.is_file()]
+        unique_names = list(set([filepath_without_extensions(_) for _ in all_files]))
```

Comment:
Typically, as far as I can recall, underscore variables are used to indicate, that you don't care about the value - which here you do. So better give it a name like `file` or `f`

Reply:
done